tests/Doctrine/Tests/ORM/Tools/Pagination/WhereInWalkerTest.php

```diff
@@ -21,9 +21,10 @@ public function testDqlQueryTransformation(string $dql, string $expectedSql): vo
         $query->setHint(Query::HINT_CUSTOM_TREE_WALKERS, [WhereInWalker::class]);
         $query->setHint(WhereInWalker::HINT_PAGINATOR_HAS_IDS, true);
 
-        $result = (new Parser($query))->parse();
+        $result   = (new Parser($query))->parse();
+        $executor = $result->getSqlFinalizer()->createExecutor($query);
```

Comment:
At this point I'm stopping, since I just realized this might be a BC break.

When not using a query hint for the output walker, the new default is `SqlOutputWalker` (instead of `SqlWalker`). This is something internal to the `Parser`, but it also makes a difference for the `ParserResult` that is returned: With the new `SqlOutputWalker`, you cannot use `ParserResult::getSqlExecutor()`, but you have to use `ParserResult::getSqlFinalizer()` and pass the `Query` to the finalizer again in order to get the `SqlExecutor`.